rand_distr/src/normal.rs

```diff
@@ -186,16 +209,54 @@ where F: Float, StandardNormal: Distribution<F>
 impl<F> LogNormal<F>
 where F: Float, StandardNormal: Distribution<F>
 {
-    /// Construct a new `LogNormal` distribution with the given mean
-    /// and standard deviation of the logarithm of the distribution.
+    /// Construct, from (log-space) mean and standard deviation
+    ///
+    /// Parameters are the "standard" log-space measures (these are the mean
+    /// and standard deviation of the logarithm of samples):
+    ///
+    /// -   `mu` (`μ`, unrestricted) is the mean of the underlying distribution
+    /// -   `sigma` (`σ ≥ 0`) is the standard deviation of the
+    ///     underlying Normal distribution
     #[inline]
-    pub fn new(mean: F, std_dev: F) -> Result<LogNormal<F>, Error> {
-        if !(std_dev >= F::zero()) {
-            return Err(Error::StdDevTooSmall);
+    pub fn new(mu: F, sigma: F) -> Result<LogNormal<F>, Error> {
+        let norm = Normal::new(mu, sigma)?;
+        Ok(LogNormal { norm })
+    }
+
+    /// Construct, from (linear-space) mean and coefficient of variation
+    ///
+    /// Parameters are linear-space measures:
+    ///
+    /// -   mean (`μ > 0`) is the (real) mean of the distribution
+    /// -   coefficient of variation (`cv = σ / μ`, requiring `cv ≥ 0`) is a
+    ///     standardized measure of dispersion
+    ///
+    /// As a special exception, `μ = 0, cv = 0` is allowed (samples are `-inf`).
+    #[inline]
+    pub fn from_mean_cv(mean: F, cv: F) -> Result<LogNormal<F>, Error> {
+        if cv == F::zero() {
+            let mu = mean.ln();
+            let norm = Normal::new(mu, F::zero()).unwrap();
+            return Ok(LogNormal { norm });
         }
-        Ok(LogNormal {
-            norm: Normal::new(mean, std_dev).unwrap(),
-        })
+        if !(mean > F::zero()) {
+            return Err(Error::MeanTooSmall);
+        }
+        if !(cv >= F::zero()) {
+            return Err(Error::BadVariance);
+        }
+
+        // Using X ~ lognormal(μ, σ), CV² = Var(X) / E(X)²
+        // E(X) = exp(μ + σ² / 2) = exp(μ) × exp(σ² / 2)
+        // Var(X) = exp(2μ + σ²)(exp(σ²) - 1) = E(X)² × (exp(σ²) - 1)
+        // but Var(X) = (CV × E(X))² so CV² = exp(σ²) - 1
+        // thus σ² = log(CV² + 1)
+        // and exp(μ) = E(X) / exp(σ² / 2) = E(X) / sqrt(CV² + 1)
+        let a = F::one() + cv * cv; // e
+        let mu = (mean / a.sqrt()).ln();
```

Comment:
`0.5 * (mean*mean / a).ln()` is probably more efficient. (Not that it matters here.)

Reply:
This causes one of the exact-equality tests to fail due to loss of accuracy. Should we care about that?

Reply:
How large is the loss of accuracy?

In this case, I would probably prefer accuracy over performance.

Reply:
It causes this to fail:
```rust
        let e = std::f64::consts::E;
        let lnorm = LogNormal::from_mean_cv(e.sqrt(), (e - 1.0).sqrt()).unwrap();
        assert_eq!((lnorm.norm.mean, lnorm.norm.std_dev), (0.0, 1.0));
```
```
thread 'normal::tests::test_log_normal_cv' panicked at 'assertion failed: `(left == right)`
  left: `(0.00000000000000011102230246251564, 1.0)`,
 right: `(0.0, 1.0)`', rand_distr/src/normal.rs:344:9
```

Reply:
That's only one ULP of difference (i.e. 2^-53), so I think this is acceptable.